src/cocotb/share/lib/vpi/VpiCbHdl.cpp

```diff
@@ -277,13 +277,24 @@ int VpiSignalObjHdl::initialise(const std::string &name,
                         return -1;
                     }
                 } else {
-                    vpi_get_value(vpi_handle(vpiLeftRange, hdl), &val);
+                    vpiHandle leftRange = vpi_handle(vpiLeftRange, hdl);
                     check_vpi_error();
-                    m_range_left = val.value.integer;
-
-                    vpi_get_value(vpi_handle(vpiRightRange, hdl), &val);
+                    vpiHandle rightRange = vpi_handle(vpiRightRange, hdl);
                     check_vpi_error();
-                    m_range_right = val.value.integer;
+
+                    if (leftRange != NULL and rightRange != NULL) {
+                        vpi_get_value(leftRange, &val);
+                        m_range_left = val.value.integer;
+
+                        vpi_get_value(rightRange, &val);
+                        m_range_right = val.value.integer;
+                    } else {
+                        LOG_WARN(
+                            "VPI: Cannot discover range bounds, guessing based "
+                            "on elements");
+                        m_range_left = 0;
+                        m_range_right = m_num_elems - 1;
+                    }
```

Comment:
What's the purpose of this branch? It's not covered by the tests.

Reply:
Sorry, I'm not following.  Without this change, the test still passes, but I get this noise:
```
     0.00ns WARNING  gpi                                VPI: Xcelium reports undefined parameters as vpiUndefined, guessing this is a logic vector
     0.00ns INFO     gpi                                VPI error
     0.00ns INFO     gpi                                NULL object vpiHandle supplied to routine vpi_get_value().
     0.00ns INFO     gpi                                VPI error
     0.00ns INFO     gpi                                NULL object vpiHandle supplied to routine vpi_get_value().
     0.00ns DEBUG    gpi                                VPI: Indexable object initialized with range [0:0] and length >32<
     0.00ns DEBUG    gpi                                VPI: has_no_default initialized with 32 elements
     0.00ns DEBUG    gpi                                VPI: Created GPI object from type vpiParameter(41)
```

Are you suggesting to do the deduction in `handle.py` instead of here?

Reply:
Nvm, [it is being covered](https://app.codecov.io/gh/cocotb/cocotb/pull/3558?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=cocotb). There's just stale warnings being displayed.